src/NServiceBus.Core/Pipeline/MutateInstanceMessage/MutateIncomingMessageBehavior.cs

```diff
@@ -15,7 +15,9 @@ class MutateIncomingMessageBehavior : Behavior<IIncomingLogicalMessageContext>
             var mutatorContext = new MutateIncomingMessageContext(current, context.Headers);
             foreach (var mutator in context.Builder.BuildAll<IMutateIncomingMessages>())
             {
-                await mutator.MutateIncoming(mutatorContext).ConfigureAwait(false);
+                await mutator.MutateIncoming(mutatorContext)
```

Comment:
I tend to agree with @danielmarbach. While of course concrete method name is a helpful information, it seems to come at quite a cost we have to pay on the "happy path" which concerns me.

I'm good with trying an approach which doesn't rely on reflection (and provides a bit less information). If we see that doesn't help enough, we should use the `Func<string>` approach.